test/AutoDiff/sil_differentiability_witness.sil

```diff
@@ -74,7 +74,7 @@ sil_differentiability_witness hidden [parameters 0 1] [results 0] [where τ_0_0
 }
 
 // CHECK-LABEL: // differentiability witness for generic
-// CHECK: sil_differentiability_witness hidden [parameters 0 1] [results 0] [where τ_0_0 : _Differentiable] @generic : $@convention(thin) <τ_0_0> (@in_guaranteed τ_0_0, Float) -> @out τ_0_0 {
+// CHECK: sil_differentiability_witness hidden [parameters 0 1] [results 0] [where T : _Differentiable] @generic : $@convention(thin) <T> (@in_guaranteed T, Float) -> @out T {
```

Comment:
`τ_0_0` feels better than `T` because 1) it is what is parsed and 2) SILFunction's use `τ_0_0` when they come out of SILGen.

Reply:
Actually, SILGen preserves `T`, so let's preserve it.
```swift
func foo<T>(_ x: T) {}
```

```console
$ swiftc -emit-silgen foo.swift
...
// foo<A>(_:)
sil hidden [ossa] @$s3fooAAyyxlF : $@convention(thin) <T> (@in_guaranteed T) -> () {
// %0                                             // user: %1
bb0(%0 : $*T):
  debug_value_addr %0 : $*T, let, name "x", argno 1 // id: %1
  %2 = tuple ()                                   // user: %3
  return %2 : $()                                 // id: %3
} // end sil function '$s3fooAAyyxlF'
```

[This comment](https://github.com/apple/swift/pull/27652/files#r335202642) has some more context.